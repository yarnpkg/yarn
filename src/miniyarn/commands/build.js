import {flags} from '@manaflair/concierge';

export default concierge =>
  concierge
    .command(`build [... args]`)
    .describe(`Run the project build script, if available`)
    .flag(flags.PROXY_COMMAND)
    .action(async args => {
      return await concierge.run(null, [`run`, `build`, ...args.args], args);
    });
