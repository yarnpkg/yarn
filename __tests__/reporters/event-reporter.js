/* @flow */

import EventReporter from '../../src/reporters/event-reporter.js';
import build from './_mock.js';

type Events = Array<Object>;

const getBuff = build(
  EventReporter,
  (data, reporter: EventReporter, events: Events): Events => events,
  (reporter: EventReporter): Events => {
    const events: Events = [];
    reporter.emit = (type: string, data: any) => {
      events.push({type, data});
    };
    return events;
  },
);

test('EventReporter.finished', async () => {
  expect(await getBuff((r) => {
    r.footer(false);
  })).toMatchSnapshot();
});

test('EventReporter.step', async () => {
  expect(await getBuff((r) => {
    r.step(1, 5, 'foobar');
  })).toMatchSnapshot();
});

test('EventReporter.log', async () => {
  expect(await getBuff((r) => {
    r.log('foobar');
  })).toMatchSnapshot();
});

test('EventReporter.success', async () => {
  expect(await getBuff((r) => {
    r.success('foobar');
  })).toMatchSnapshot();
});

test('EventReporter.error', async () => {
  expect(await getBuff((r) => {
    r.error('foobar');
  })).toMatchSnapshot();
});

test('EventReporter.info', async () => {
  expect(await getBuff((r) => {
    r.info('foobar');
  })).toMatchSnapshot();
});

test('EventReporter.command', async () => {
  expect(await getBuff((r) => {
    r.command('foobar');
  })).toMatchSnapshot();
});

test('EventReporter.warn', async () => {
  expect(await getBuff((r) => {
    r.warn('foobar');
  })).toMatchSnapshot();
});
