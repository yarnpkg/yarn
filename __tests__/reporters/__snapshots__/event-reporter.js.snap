exports[`test EventReporter.command 1`] = `
Array [
  Object {
    "data": "foobar",
    "type": "command",
  },
]
`;

exports[`test EventReporter.error 1`] = `
Array [
  Object {
    "data": "foobar",
    "type": "error",
  },
]
`;

exports[`test EventReporter.finished 1`] = `
Array [
  Object {
    "data": 0,
    "type": "finished",
  },
]
`;

exports[`test EventReporter.info 1`] = `
Array [
  Object {
    "data": "foobar",
    "type": "info",
  },
]
`;

exports[`test EventReporter.log 1`] = `
Array [
  Object {
    "data": "foobar",
    "type": "log",
  },
]
`;

exports[`test EventReporter.step 1`] = `
Array [
  Object {
    "data": Object {
      "current": 1,
      "message": "foobar",
      "total": 5,
    },
    "type": "step",
  },
]
`;

exports[`test EventReporter.success 1`] = `
Array [
  Object {
    "data": "foobar",
    "type": "success",
  },
]
`;

exports[`test EventReporter.warn 1`] = `
Array [
  Object {
    "data": "foobar",
    "type": "warning",
  },
]
`;
